pkg/workloads/spark_job/spark_util.py

```diff
@@ -569,22 +650,33 @@ def validate_transformer(column_name, df, ctx, spark):
                         "{} != {}".format(ts_row, tp_row),
                     )
 
+    if transformer["output_type"] == "unknown":
```

Comment:
I think we should move this to the top of the section above, since it's better to check types before checking values (otherwise the value check will fail and the error message might be confusing). E.g. delete lines 653 - 665, and add this at line 631 after `if hasattr(trans_impl, "transform_spark") and hasattr(trans_impl, "transform_python"):`:

```
if inferred_spark_type != inferred_python_type:
    raise UserRuntimeException(
        "transformed column " + column_name,
        "type inference failed, transform_spark and transform_python had differing types.",
        "transform_python: " + inferred_python_type,
        "transform_spark: " + inferred_spark_type,
    )
```

(they'll either both be `None` or not, so I think the `if` check is sufficient)